misc/make-deps.sh

```diff
@@ -102,10 +102,11 @@ go get github.com/blynn/nex				# for lexing
 go get golang.org/x/tools/cmd/goyacc			# formerly `go tool yacc`
 go get golang.org/x/tools/cmd/stringer			# for automatic stringer-ing
 go get golang.org/x/lint/golint				# for `golint`-ing
+go get golang.org/x/tools/cmd/goimports		# for fmt
 go get github.com/tmthrgd/go-bindata/go-bindata	# for compiling in non golang files
 if env | grep -q -e '^TRAVIS=true$' -e '^JENKINS_URL=' -e '^BUILD_TAG=jenkins'; then
 	go get -u gopkg.in/alecthomas/gometalinter.v1 && mv "$(dirname $(command -v gometalinter.v1))/gometalinter.v1" "$(dirname $(command -v gometalinter.v1))/gometalinter" && gometalinter --install	# bonus
 fi
-command -v mdl &>/dev/null || gem install mdl --no-document || true	# for linting markdown files
-command -v fpm &>/dev/null || gem install fpm --no-document || true	# for cross distro packaging
+command -v mdl &>/dev/null || $sudo_command gem install mdl --no-document || true	# for linting markdown files
+command -v fpm &>/dev/null || $sudo_command gem install fpm --no-document || true	# for cross distro packaging
```

Comment:
Is root really necessary? For linux it seems it is not. If it is for OSX, can we wrap the sudo variants in an `if OSX` sort of thing please?

Reply:
This is for Linux, this is what i get on Ubuntu: 
```
Fetching: mdl-0.5.0.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /var/lib/gems/2.5.0 directory.
Fetching: fpm-1.11.0.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /var/lib/gems/2.5.0 directory.
~/g/s/g/p/mgmt (master|✔) $ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
An alternative is to use `--user` but often this requires modification to `PATH` as most users don't have Ruby setup.

Reply:
Interesting! Let's to the --user option if that will work, since the gem stuff is only used for package stuff and making docs, so most users won't need it, and causing less root stuff on their machines is probably a good thing.

Reply:
Agree. The commands don't fail the dep install anyways, but can be a red herring for new (or recurring) users.